### PR TITLE
Update prom-client to v15.1.0

### PR DESCRIPTION
**Pull request by p-kuen**
BREAKING CHANGE: this drops support for node v16 as prom-client also dropped support for node 16.

Node 16 is out of support since 11 Sep 2023.

prom-client adds support for opentelemetry. This PR just changes all code so that it works the same as before. Maybe opentelemetry support as requested in #82 should not be a big problem anymore.

### CODE REVIEW

File: src/__tests__/edge-cases.spec.ts [hunk 1]
```diff
@@ -144,7 +144,7 @@ describe('edge cases', () => {
           expect.stringMatching(
             /process_cpu_user_seconds_total\{foo="bar"\} \d+/
           ),
-          'http_request_duration_seconds_count{method="GET",route="/test",status_code="200",foo="bar"} 1',
+          'http_request_duration_seconds_count{foo="bar",method="GET",route="/test",status_code="200"} 1',
```
Comment (p-kuen) [inline src/__tests__/edge-cases.spec.ts:147]: For some reason prom-client changed the order of the properties, so I had to change them here.

File: src/types.ts [hunk 2]
```diff
@@ -33,7 +33,7 @@ export interface IMetricsRouteContextConfig {
  * @see {@link https://github.com/siimon/prom-client#default-metrics | prom-client} for extra options
  */
 export interface IDefaultMetricsConfig
-  extends DefaultMetricsCollectorConfiguration {
+  extends DefaultMetricsCollectorConfiguration<'text/plain; version=0.0.4; charset=utf-8'> {
```
Comment (p-kuen) [inline src/types.ts:36]: prom-client requires a generic parameter here because of opentelemetry support. So I added the prometheus way for now.
